config/configgrpc/configgrpc.go

```diff
@@ -311,3 +333,11 @@ func GetGRPCCompressionKey(compressionType string) string {
 	}
 	return CompressionUnsupported
 }
+
+func RegisterClientDialOptionHandlers(handlers ...ClientDialOptionHandler) {
+	mu.Lock()
```

Comment:
why do we need mutex here?

Reply:
Can we guarantee that this func is called only from a single thread?

Reply:
Removed the mutexes.